### PR TITLE
LOK-2474 BE: BFF: Monitoring policies: DownloadToCsv

### PR DESCRIPTION
**Pull request by Mohammad-Junid**
## Description
Add refresh and CSV download for new Monitoring Policies table.

## Jira link(s)
- https://opennms.atlassian.net/browse/LOK-2474

## Checklist
* [ ] Follows Lōkahi's [development guidelines.](https://github.com/OpenNMS-Cloud/lokahi/wiki/Development-Guidelines)
* [ ] Appropriate reviewer(s) have been selected.
* [ ] Jira issue(s) have been updated to "In Review".
* [ ] Includes [appropriate tests.](https://github.com/OpenNMS-Cloud/lokahi/wiki/Test-Strategy)
* [ ] Documentation has been updated as necessary.
* [ ] Notify devops of changes to the Charts.
* [ ] Notify documentation team of any changes to names of screens or features (affects URLs).

### CODE REVIEW

File: rest-server/src/main/java/org/opennms/horizon/server/model/inventory/DownloadResponse.java
Comment (smunir-onms) [inline rest-server/src/main/java/org/opennms/horizon/server/model/inventory/DownloadResponse.java:1]: A cool suggestion: DownloadResponse class should be in the model package (also inventory.DownloadFormat should be moved to the model package).
The following download responses have the same properties, so these repeating models should be removed and replaced with this model;

TopNResponse
DownloadAlertsResponse
IpInterfaceResponse
SnmpInterfaceResponse
SearchEventsResponse

File: rest-server/src/main/java/org/opennms/horizon/server/service/GraphQLAlertService.java [hunk 1]
```diff
@@ -149,6 +152,26 @@ public Flux<MonitorPolicy> listMonitoryPolicies(@GraphQLEnvironment ResolutionEn
         return Flux.fromIterable(alertsClient.listMonitorPolicies(headerUtil.getAuthHeader(env)));
     }
 
+    @GraphQLQuery
+    public Mono<DownloadResponse> searcbAndDownloadMonitoryPolicies(
```
Comment (smunir-onms) [inline rest-server/src/main/java/org/opennms/horizon/server/service/GraphQLAlertService.java:156]: rename to searcbAndDownloadMonitoringPolicies

File: alert/src/main/java/org/opennms/horizon/alertservice/grpc/MonitorPolicyGrpc.java [hunk 2]
```diff
@@ -199,6 +205,28 @@ public void countAlertByRuleId(Int64Value request, StreamObserver<Int64Value> re
                         () -> responseObserver.onError(StatusProto.toStatusRuntimeException(badTenant())));
     }
 
+    @Override
+    public void searchMonitorPolicies(
```
Comment (smunir-onms) [inline alert/src/main/java/org/opennms/horizon/alertservice/grpc/MonitorPolicyGrpc.java:209]: rename to searchMonitoringPolicies

File: rest-server/src/main/java/org/opennms/horizon/server/service/GraphQLAlertService.java [hunk 3]
```diff
@@ -254,6 +277,39 @@ public Mono<DownloadAlertsResponse> downloadRecentAlertsByNode(
         }
     }
 
+    @GraphQLQuery(name = "getNodesCountByMonitoringPolicy")
+    public Mono<Long> getNodesCountByMonitoringPolicy(
+            @GraphQLArgument(name = "id") long id, @GraphQLEnvironment ResolutionEnvironment env) {
+        return Mono.just(alertsClient.getNodesCountByMonitoringPolicy(id, headerUtil.getAuthHeader(env)));
+    }
+
+    private DownloadResponse generateDownloadableMonitorPoliciesResponse(
```
Comment (smunir-onms) [inline rest-server/src/main/java/org/opennms/horizon/server/service/GraphQLAlertService.java:286]: rename to generateDownloadableMonitoringPoliciesResponse

File: rest-server/src/main/java/org/opennms/horizon/server/service/grpc/AlertsClient.java [hunk 4]
```diff
@@ -34,23 +34,7 @@
...
-import org.opennms.horizon.alerts.proto.EventType;
-import org.opennms.horizon.alerts.proto.Filter;
-import org.opennms.horizon.alerts.proto.ListAlertEventDefinitionsRequest;
-import org.opennms.horizon.alerts.proto.ListAlertsRequest;
-import org.opennms.horizon.alerts.proto.ListAlertsResponse;
-import org.opennms.horizon.alerts.proto.MonitorPolicyProto;
-import org.opennms.horizon.alerts.proto.MonitorPolicyServiceGrpc;
-import org.opennms.horizon.alerts.proto.Severity;
-import org.opennms.horizon.alerts.proto.TimeRangeFilter;
+import org.opennms.horizon.alerts.proto.*;
```
Comment (smunir-onms) [inline rest-server/src/main/java/org/opennms/horizon/server/service/grpc/AlertsClient.java:37]: avoid wildcard imports, import specific classes individually

File: rest-server/src/main/java/org/opennms/horizon/server/service/GraphQLAlertService.java (hunk 1 above)
Comment (Mohammad-Junid) [reply rest-server/src/main/java/org/opennms/horizon/server/service/GraphQLAlertService.java:156]: Done

File: rest-server/src/main/java/org/opennms/horizon/server/service/GraphQLAlertService.java (hunk 3 above)
Comment (Mohammad-Junid) [reply rest-server/src/main/java/org/opennms/horizon/server/service/GraphQLAlertService.java:286]: Done

File: rest-server/src/main/java/org/opennms/horizon/server/model/inventory/DownloadResponse.java
Comment (aurang-zeb313) [reply rest-server/src/main/java/org/opennms/horizon/server/model/inventory/DownloadResponse.java:1]: We need to create two task, one for BFF and one for BE

**Review by smunir-onms — COMMENTED** [review]
LGTM, just add screenshot of graphQL query result

**Comment by Mohammad-Junid** [comment]
image

**Review by fooker — APPROVED** [review]

**Pull request merged**